src/java/org/apache/cassandra/db/compaction/unified/Controller.java

```diff
@@ -268,9 +268,10 @@ public int getNumShards(double localDensity)
         if (minSSTableSize > 0)
         {
             double count = localDensity / minSSTableSize;
-            // Minimum size only applies if it is smaller than the base count.
+            // Minimum size only applies if the base count would result in smaller sstables.
+            // We also want to use the min size if we don't yet know the flush size (density is NaN).
             // Note: the minimum size cannot be larger than the target size's minimum.
-            if (count < baseShardCount)
+            if (!(count >= baseShardCount)) // also true for count == NaN
```

Comment:
Tests typically run with unknown flush size and we want them to use just one shard.

It also generally makes better sense to start non-splitting if we don't know what size we should expect.